library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileOutputOperator.java

```diff
@@ -484,34 +484,40 @@ public FSFilterStreamContext load(@Nonnull String filename)
         //of this operator.
         for(String seenFileName: endOffsets.keySet()) {
           try {
-            Integer part = openPart.get(seenFileName).getValue() + 1;
+            Integer fileOpenPart = this.openPart.get(seenFileName).getValue();
+            int nextPart = fileOpenPart + 1;
```

Comment:
1. I didn't notice that code. Thats great. However there is some ambiguity here. For example startsWith(fileName) condition could match for tmp files of filename.10 when looking for tmp files of filename.1. Also since the user can determine how to form the part filename looking for something like startsWith(fileName + ".") is not always guaranteed either as I could choose my part 10 filename to be filename.1.0, will have to look for pattern filename.<long>.tmp specifically.

Regd 2 it could take some time for namenode to realize client is dead. Not related to earlier issue but since there was no proper close and since there cant be multiple writes to same file I am guessing there could be an issue. Will be good to try it out.